server/src/main/java/org/opensearch/rest/RestHandler.java

```diff
@@ -163,6 +164,12 @@ public ReplacedRoute(Method method, String path, Method deprecatedMethod, String
             this.deprecatedPath = deprecatedPath;
         }
 
+        public ReplacedRoute(Route route, String prefix, String deprecatedPrefix) {
+            super(route.getMethod(), prefix + route.getPath());
```

Comment:
```suggestion
            this(route.getMethod, prefix + route.getPath(), deprecatedPrefix + route.getPath()
```
and introduce 2 ctor.

Reply:
new ctor added in the latest commit.